app/src/main/java/currencyConverter/service/CurrencyConverter.java

```diff
@@ -0,0 +1,22 @@
+package currencyConverter.service;
+
+import currencyConverter.exception.CurrencyRateFetchingException;
+import currencyConverter.model.CurrencyCode;
+
+public class CurrencyConverter implements ICurrencyConverter {
+
+    private static final String RESULT_FORMAT_PATTERN = "%.2f %s";
+
+    private ICurrencyRatesProvider currencyRatesProvider;
+
+    public CurrencyConverter() {
+        this.currencyRatesProvider = new CurrencyRatesProvider();
+    }
+
+    @Override
+    public String convert(CurrencyCode from, CurrencyCode to, double value) throws CurrencyRateFetchingException {
```

Comment:
I don't really like that converter dictats me the format of converting result. As variant, we can initialize Converter with specific format, using yours one by default. And it should have at least two methods - to return raw double value and formatted string, if neccessary.

Reply:
I'll better add an overloaded method, that returns double, if you think you need it. 
Dont't think that a big amount of interfaces and services that do the same thing (but return different formats) is a good idea.
Or did you mean make CurrencyConverter to support generics? If so, there will be a little overhead for results casting, so the overloaded methods are the best solution in this case I think.

Reply:
No, I didn't mean generics. Neither I wanted multiple interfaces. But different methods, or possibility to configurate is exactly required. What If we decide later to output result of conversion in Romanian numerals, drown by curves with OpenGL ? Should we convert your string back to double again? Let's make CurrencyConverter a simple calculator which multiples several values and return result value, but format of representation should be chosen by clients of Converter.